Fix build script error handling on windows

brson opened the pull request:
This gets past the error reported in https://github.com/nikomatsakis/lalrpop/issues/18, though I see further parser failures in the calculator example.

nikomatsakis commented:
Thanks Brian. What are the other failures you see? I'll have to double check but I think it passes locally.

brson commented:
@nikomatsakis 

```
$ cargo build
   Compiling lalrpop-snap v0.6.1 (file:///C:/Users/brian/Documents/dev/lalrpop/lalrpop-snap)
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\build/mod.rs:83:35: 83:51 warning: unused variable: `lalrpop_metadata`, #[warn(unused_variables)] on by default
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\build/mod.rs:83     fn compare_modification_times(lalrpop_metadata: &fs::Metadata,
                                                                                                        ^~~~~~~~~~~~~~~~
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\build/mod.rs:84:35: 84:46 warning: unused variable: `rs_metadata`, #[warn(unused_variables)] on by default
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\build/mod.rs:84                                   rs_metadata: &fs::Metadata)
                                                                                                        ^~~~~~~~~~~
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\lr1\ascent.rs:551:39: 551:52 warning: use of deprecated item: renamed to join, #[warn(deprecated)] on by default
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\lr1\ascent.rs:551               let_name, pattern_names.connect(", "));
                                                                                                              ^~~~~~~~~~~~~
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\rust/mod.rs:9:47: 9:70 note: in this expansion of format_args!
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\rust/mod.rs:9:9: 9:73 note: in this expansion of try! (defined in <std macros>)
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\lr1\ascent.rs:550:9: 551:54 note: in this expansion of rust! (defined in C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\rust/mod.rs)
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\normalize\lower/mod.rs:259:31: 259:44 warning: use of deprecated item: renamed to join, #[warn(deprecated)] on by default
C:\Users\brian\Documents\dev\lalrpop\lalrpop-snap\src\normalize\lower/mod.rs:259                     name_strs.connect(", ")
                                                                                                               ^~~~~~~~~~~~~
   Compiling lalrpop v0.5.0
failed to run custom build command for `lalrpop v0.5.0`
Process didn't exit successfully: `C:\Users\brian\Documents\dev\lalrpop\doc\calculator\target\debug\build\lalrpop-f7721ad1ac348df2\build-script-build` (exit code: 1)
--- stdout
src\parser\lrgrammar.lalrpop:45:47: 45:47 error: unexpected token: `;`
      <l:Lifetime> => TypeParameter::Lifetime(l);
                                                ^
```

nikomatsakis commented:
@brson can you give it another try? I tweaked the `Cargo.toml` files.

brson commented:
I had lalrpop-snap overridden locally, probably with a bogus revision.
